docs/maintainers/adr/0007-bind-implicit-language-to-twirl-conponent.md

```diff
@@ -0,0 +1,69 @@
+# Populate language parameter from implicit messages
+
+* Status: accepted
+* Date: 2020-03-11
+
+Technical Story: PLATUI-531
+
+## Context and Problem Statement
+
+In [ADR-0001](0001-play-frontend-hmrc-mirrors-hmrc-frontend.md) we established
+the principle that play-frontend-hmrc should mirror hmrc-frontend
+in providing a pure Twirl port of the underlying hmrc-frontend Nunjucks library. We decided that the only
+exception to that would be a helpers package containing wrappers that enhance the underlying components with
+platform and Play framework aware functionality.
+
+One common enhancement has been to populate the language parameter found on many of hmrc-frontend's
+components with the language associated with the Messages object in implicit scope. Requiring consuming
+services to manually provide this parameter is a form of repetitious boilerplate we are actively
+aiming to minimise.
+
+In PLATUI-531, a story to develop a version of govukCharacterCount supporting the Welsh
+language, we faced the need to create and maintain a wrapper for hmrcCharacterCount whose sole purpose would be to supply
+this parameter. Given [ADR-0003](0003-use-the-suffix-helpers-for-helper-components.md), this wrapper would have to be named
+hmrcCharacterCountHelper, a rather non-intuitive name bearing in mind we would generally be expecting
+teams to use this component rather than the more appropriately named hmrcCharacterCount.
+
+We therefore made a joint decision to allow as an exception the passing through of an implicit Messages to Twirl
+components, making them Play framework aware, as a pragmatic exception to allow us to maintain our current testing
+strategy whilst not makng a confusing experience for service teams using the library.
+
+## Decision Drivers
+
+* the need for component naming to be unsurprising and intuitive
+* the need for components to justify their cost of maintenance
+* play-frontend-hmrc should be Play-framework aware and not require manual wiring of parameters
+that can be found in implicit scope
+* the importance of maintaining our test strategy and for it not to be undermined.
+
+## Considered Options
+
+* Allow components to take an implicit Messages object and used to populate the language parameter
+* Do nothing, i.e. continue to use the pattern of helpers even when the only additional parameter
+is the implicit languauge
+
+## Decision Outcome
+
+Chosen option: "Allow components to take an implicit Messages object", because we can do this without
+undermining our test strategy and doing so will improve the developer experience.
+
+### Positive Consequences
```

Comment:
Seems like the negative consequences outweigh the positives. Should we have as additional positives:
* Teams have available to them a single component that is appropriately named
* The new component has an idiomatic interface that makes full use of the Play framework's internationalisation features.